src/symbolic/symbolic_memory.ml

```diff
@@ -154,7 +154,8 @@ let extract v pos =
     value (Num (I8 i'))
   | Cvtop (_, Zero_extend 24, ({ node = Symbol _; _ } as sym))
   | Cvtop (_, Sign_extend 24, ({ node = Symbol _; _ } as sym))
-    when ty sym = Ty_bitv 8 ->
+    when match ty sym with Ty_bitv 8 -> true | _ -> false ->
+    (* TODO: implement an equal function in smtml for this *)
```

Comment:
cc @filipeom, do you think you could add such a function ?

Reply:
A equality function for `Ty.t` ? We have [this](https://github.com/formalsec/smtml/blob/a3c524421082acf4c76e5dcc83e9ea133bdd2264/lib/ty.ml#L307), would it suffice or do we need something better?

Reply:
Indeed.. Thanks!